02 - Web UI Template/CognitiveSearch.UI/Search/DocumentSearchClient.cs

```diff
@@ -370,33 +376,37 @@ private string[] GetContainerSasUris()
-
-            SharedAccessBlobPolicy adHocPolicy = new SharedAccessBlobPolicy()
-            {
-                SharedAccessExpiryTime = DateTime.UtcNow.AddHours(24),
-                Permissions = SharedAccessBlobPermissions.Read
-            };
-
+            StorageSharedKeyCredential storageSharedKeyCredential = new StorageSharedKeyCredential(accountName, accountKey);
             containerAddresses[0] = _configuration.GetSection("StorageContainerAddress")?.Value.ToLower();
-            CloudBlobContainer container = new CloudBlobContainer(new Uri(containerAddresses[0]), new StorageCredentials(accountName, accountKey));
-            tokens[0] = container.GetSharedAccessSignature(adHocPolicy, null);
-
-            // Get token for second indexer data source
             containerAddresses[1] = _configuration.GetSection("StorageContainerAddress2")?.Value.ToLower();
+            containerAddresses[2] = _configuration.GetSection("StorageContainerAddress3")?.Value.ToLower();
+            int containerAddressesLength = containerAddresses.Length;
             if (!String.Equals(containerAddresses[1], defaultContainerUriValue))
             {
-                CloudBlobContainer container2 = new CloudBlobContainer(new Uri(containerAddresses[1]), new StorageCredentials(accountName, accountKey));
-                tokens[1] = container2.GetSharedAccessSignature(adHocPolicy, null);
+                containerAddressesLength--;
             }
-
-            // Get token for third indexer data source
-            containerAddresses[2] = _configuration.GetSection("StorageContainerAddress3")?.Value.ToLower();
             if (!String.Equals(containerAddresses[2], defaultContainerUriValue))
             {
-                CloudBlobContainer container3 = new CloudBlobContainer(new Uri(containerAddresses[2]), new StorageCredentials(accountName, accountKey));
-                tokens[2] = container3.GetSharedAccessSignature(adHocPolicy, null);
+                containerAddressesLength--;
             }
+            for (int i = 0; i < containerAddressesLength; i++) {
+                BlobContainerClient container = new BlobContainerClient(new Uri(containerAddresses[i]), new StorageSharedKeyCredential(accountName, accountKey));
+                var policy = new BlobSasBuilder
+                {
+                    Protocol = SasProtocol.HttpsAndHttp,
+                    BlobContainerName = container.Name,
+                    Resource = "c",
+                    StartsOn = DateTimeOffset.UtcNow,
+                    ExpiresOn = DateTimeOffset.UtcNow.AddHours(24),
+                    IPRange = new SasIPRange(IPAddress.None, IPAddress.None)
+                };
+                policy.SetPermissions(BlobSasPermissions.Read);
+                var sas = policy.ToSasQueryParameters(storageSharedKeyCredential).ToString();
+                UriBuilder sasUri = new UriBuilder(container.Uri);
```

Comment:
nit: can use `BlobUriBuilder` and then set `Sas` property to `policy.ToSasQueryParameters` instead of using `UriBuilder` and setting `Query` property.